src/WebJobs.Script/Description/DotNet/FunctionMetadataResolver.cs

```diff
@@ -179,10 +180,25 @@ private ImmutableArray<PortableExecutableReference> GetMetadataFromReferencePath
             }
             else
             {
-                // Treat the reference as a private assembly reference
-                string filePath = Path.Combine(_privateAssembliesPath, reference);
-                if (File.Exists(Path.Combine(filePath)))
+                bool externalReference = false;
+                string basePath = _privateAssembliesPath;
+                
+                // If this is is a relative assembly reference, use the function directory as the base probing path
+                if (reference.IndexOfAny(new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar }) > -1)
+                {
+                    basePath = Path.GetDirectoryName(_functionMetadata.ScriptFile);
+                    externalReference = true;
+                }
+
+                string filePath = Path.GetFullPath(Path.Combine(basePath, reference));
+                if (File.Exists(filePath))
```

Comment:
I think it would be better to hard fail on the missing assembly than ignoring it and then letting the code compile (potentially) fail as a result. It's like using `csc /r foo.dll`. If it can't find it, it ails right there.